projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -310,51 +322,56 @@ export class DatagridComponent<R> implements OnInit, AfterViewInit, OnDestroy {
     }
 
     /**
-     * Sets the button configuration on the datagrid.
-     *
-     * {@link ButtonConfig.inactiveDisplayMode} defualts to Disabled.
-     */
-    @Input() set buttonConfig(config: ButtonConfig<R>) {
-        this._buttonConfig = config;
-        this._buttonConfig.contextualButtonConfig = this._buttonConfig.contextualButtonConfig
-            ? this._buttonConfig.contextualButtonConfig
-            : {
-                  buttons: [],
-                  position: ContextualButtonPosition.TOP,
-              };
-        this._buttonConfig.contextualButtonConfig.buttonContents =
-            this._buttonConfig.contextualButtonConfig.buttonContents || TextIcon.ICON;
-        this._buttonConfig.inactiveDisplayMode =
-            this._buttonConfig.inactiveDisplayMode || InactiveButtonDisplayMode.Disable;
-        if (this._buttonConfig.contextualButtonConfig.featured) {
-            this.featuredButtons = new Map(
-                this._buttonConfig.contextualButtonConfig.featured.map(featuredButtonClass => [
-                    featuredButtonClass,
-                    this._buttonConfig.contextualButtonConfig.buttons.find(
-                        button => button.class === featuredButtonClass
-                    ),
-                ])
-            );
-            this.featuredButtons.forEach(featured => {
-                if (!featured) {
-                    throw new Error('Featured button was not found');
+     * Actions to be displayed on a grid or in a row
+     */
+    private _actions: ActionItem<R, unknown>[] = [];
+
+    /**
+     * List of actions given by the caller
+     */
+    @Input() set actions(actions: ActionItem<R, unknown>[]) {
+        this._actions = actions.map(action => {
+            const actionHandler: ActionHandlerType<R, unknown> = action.handler;
```

Comment:
It is not good to modify the provided element, better do :
action = {...action}